### PR TITLE
Fix for Serialization issue with OAuthInfo

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
Removed repeated calls to salesforce.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/plugin/OAuthInfo.java [hunk 1]
```diff
@@ -16,10 +16,12 @@
 
 package io.cdap.plugin.salesforce.plugin;
 
+import java.io.Serializable;
+
 /**
  * Class to carry OAuth information returned by the {@code ${oauth}} macro function.
  */
-public final class OAuthInfo {
+public final class OAuthInfo implements Serializable {
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/OAuthInfo.java]: Do we still need this to be serializable?

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java [hunk 2]
```diff
@@ -83,7 +85,15 @@ public class SalesforceConnector implements DirectConnector {
   @Override
   public void test(ConnectorContext connectorContext) throws ValidationException {
     FailureCollector collector = connectorContext.getFailureCollector();
-    config.validate(collector);
+    OAuthInfo oAuthInfo = null;
+    try {
+      oAuthInfo = Authenticator.getOAuthInfo(config.getAuthenticatorCredentials());
+    } catch (Exception e) {
+      collector.addFailure("Error encountered while establishing connection: " + e.getMessage(),
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java]: Will the exception message here have all the information? Do we need to check for `IApiFault` ?

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/actions/SalesforcePropertiesPageActions.java [hunk 3]
```diff
@@ -124,12 +126,11 @@ public static void verifyNoErrorsFoundSuccessMessage() {
   }
 
   private static AuthenticatorCredentials setAuthenticationCredentialsOfAdminUser() {
-    return new AuthenticatorCredentials(
-      PluginPropertyUtils.pluginProp("admin.username"),
+    return new AuthenticatorCredentials(null, PluginPropertyUtils.pluginProp("admin.username"),
```
Comment (greeshmaswaminathan) [inline src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/actions/SalesforcePropertiesPageActions.java]: Why is this change required?

File: src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java [hunk 4]
```diff
@@ -46,13 +46,13 @@ public AuthenticatorCredentials(String username, String password,
...
   }
 
-  private AuthenticatorCredentials(@Nullable OAuthInfo oAuthInfo,
-                                   @Nullable String username,
-                                   @Nullable String password,
-                                   @Nullable String consumerKey,
-                                   @Nullable String consumerSecret,
-                                   @Nullable String loginUrl,
-                                   @Nullable Integer connectTimeout) {
+  public AuthenticatorCredentials(@Nullable OAuthInfo oAuthInfo,
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java]: Was this made public for calling from e2e test? I am not sure why we need to call this constructor instead of the above.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java [hunk 5]
```diff
@@ -66,18 +69,37 @@ public SalesforceBatchSink(SalesforceSinkConfig config) throws ConnectionExcepti
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
-    config.validate(stageConfigurer.getInputSchema(), stageConfigurer.getFailureCollector());
+    FailureCollector collector = stageConfigurer.getFailureCollector();
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java:66]: Put the code to get OAuthInfo with collector failure in a method , so that it need not be repeated in multiple places. Can be even moved into one of the utils, if the same code is applicable to all the connectors/src/sink. Then the method call can be inlined to the validate calls.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java [hunk 6]
```diff
@@ -168,13 +169,14 @@ public String getFQN(String orgId, String sObject) {
...
-      (connection.getAuthenticatorCredentials());
+  public String getOrgId(OAuthInfo oAuthInfo) throws ConnectionException {
+    AuthenticatorCredentials credentials = new AuthenticatorCredentials(oAuthInfo,
+            this.getConnection().getConnectTimeout());
+    PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
     return partnerConnection.getUserInfo().getOrganizationId();
   }
 
-  protected void validateFilters(FailureCollector collector) {
+  public void validateFilters(FailureCollector collector) {
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java:178]: Why is this made public?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 7]
```diff
@@ -75,26 +77,50 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
-    config.validate(stageConfigurer.getFailureCollector()); // validate before macros are substituted
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateFilters(collector);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:82]: Why do we need a separate call to validateFilters() here ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 8]
```diff
@@ -75,26 +77,50 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
...
+    try {
+      oAuthInfo = Authenticator.getOAuthInfo(config.getConnection().getAuthenticatorCredentials());
+    } catch (Exception e) {
+      collector.addFailure("Error encountered while establishing connection: " + e.getMessage(),
+                      "Please verify authentication properties are provided correctly")
+              .withStacktrace(e.getStackTrace());
+      throw collector.getOrThrowException();
+    }
+    config.validate(collector, oAuthInfo);
+    config.validateFilters(collector);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:101]: can we combine all these checks? Are these not always called together?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java [hunk 9]
```diff
@@ -79,7 +81,20 @@ public SalesforceBatchSource(SalesforceSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // validate when macros not yet substituted
-    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateSOQLQueryParameters(collector);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java:85]: This looks like a new validation ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java [hunk 10]
```diff
@@ -200,7 +200,43 @@ public void validate(FailureCollector collector) {
       }
     }
     validateSchema(collector);
-    validatePKChunk(collector);
+    validatePKChunk(collector, oAuthInfo);
+  }
+  public void validateSOQLQueryParameters(FailureCollector collector) {
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java:206]: Leave a space before the method. Is this new functionality or refactored to move here?

File: src/main/java/io/cdap/plugin/salesforce/plugin/OAuthInfo.java (hunk 1 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/OAuthInfo.java]: As of now, this is not required as we are using OAuthInfo in configurePipeline and prepareRun methods only, not sending anything calls to network. Will remove this.

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java (hunk 2 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java]: Added.

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/actions/SalesforcePropertiesPageActions.java (hunk 3 above)
Comment (vikasrathee-cs) [reply src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/actions/SalesforcePropertiesPageActions.java]: Need to add connectTimeout only. removed

File: src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java (hunk 4 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java]: made it private.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java (hunk 5 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java:66]: Added a method in SalesforceConnectionUtil and moved the common code there.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java (hunk 6 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java:178]: Previously it was getting called from its sub classes but now that we have segregated the code with connection to salesforce and without connection to salesforce. We have to call the validation methods that don't require OAuthInfo.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 7 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:82]: In case of not able to connect with Salesforce, we need to basic validations which doesn't require connection to be established.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 8 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:101]: We have to segregate it based on whether it requires connection to do the validation or not.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java (hunk 9 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java:85]: This is not a new validation. We need to perform this if connection is not there and still we have to check whether soql is correct and both of sObject and SOQL query are not null, one should have some value.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java (hunk 10 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java:206]: Added. These are the validations that need to be done without setting up connection with Salesforce.

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java [hunk 11]
```diff
@@ -19,11 +19,14 @@
 import com.sforce.soap.partner.fault.IApiFault;
 import com.sforce.ws.ConnectionException;
 import com.sforce.ws.ConnectorConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
 import io.cdap.plugin.salesforce.authenticator.Authenticator;
 import io.cdap.plugin.salesforce.authenticator.AuthenticatorCredentials;
 import io.cdap.plugin.salesforce.plugin.OAuthInfo;
 import org.apache.hadoop.conf.Configuration;
 
+import javax.security.auth.login.FailedLoginException;
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java]: Looks like an unused import ?

File: src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java [hunk 12]
```diff
@@ -47,12 +47,12 @@ public AuthenticatorCredentials(String username, String password,
   }
 
   private AuthenticatorCredentials(@Nullable OAuthInfo oAuthInfo,
-                                   @Nullable String username,
-                                   @Nullable String password,
-                                   @Nullable String consumerKey,
-                                   @Nullable String consumerSecret,
-                                   @Nullable String loginUrl,
-                                   @Nullable Integer connectTimeout) {
+                                  @Nullable String username,
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java]: Looks like an unintended change. can you revert this ?

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java [hunk 13]
```diff
@@ -46,7 +46,9 @@
 import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
 import io.cdap.plugin.salesforce.SalesforceConstants;
 import io.cdap.plugin.salesforce.SalesforceSchemaUtil;
+import io.cdap.plugin.salesforce.authenticator.Authenticator;
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java]: Is this an unused import?

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java [hunk 14]
```diff
@@ -35,7 +35,11 @@
 import io.cdap.cdap.etl.api.connector.Connector;
 import io.cdap.plugin.common.Asset;
 import io.cdap.plugin.common.LineageRecorder;
+import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
 import io.cdap.plugin.salesforce.SalesforceConstants;
+import io.cdap.plugin.salesforce.authenticator.Authenticator;
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: This import may not be required. Is Authenticator being used anywhere else?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 15]
```diff
@@ -75,26 +78,36 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
...
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
-    config.validate(stageConfigurer.getFailureCollector()); // validate before macros are substituted
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateFilters(collector);
+      return;
+    }
+    OAuthInfo oAuthInfo = SalesforceConnectionUtil.getOAuthInfo(config.getConnection().getAuthenticatorCredentials(),
+            collector);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: Fix the indentation everywhere.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 16]
```diff
@@ -75,26 +78,36 @@ public SalesforceBatchMultiSource(SalesforceMultiSourceConfig config) {
...
 
   @Override
   public void prepareRun(BatchSourceContext context) throws ConnectionException {
     FailureCollector collector = context.getFailureCollector();
-    config.validate(collector);
-    config.validateSObjects(collector);
+    OAuthInfo oAuthInfo = SalesforceConnectionUtil.getOAuthInfo(config.getConnection().getAuthenticatorCredentials(),
+            collector);
+    config.validate(collector, oAuthInfo);
+    config.validateFilters(collector);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:101]: Can we do the validations that do not require the oAuthinfo first ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 17]
```diff
@@ -35,8 +35,11 @@
 import io.cdap.cdap.etl.api.batch.BatchSource;
 import io.cdap.cdap.etl.api.batch.BatchSourceContext;
 import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.plugin.salesforce.SalesforceConnectionUtil;
 import io.cdap.plugin.salesforce.SalesforceConstants;
+import io.cdap.plugin.salesforce.authenticator.Authenticator;
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: Unused import ?

Comment (greeshmaswaminathan) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: Check other classes too.

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java (hunk 11 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java]: removed.

File: src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java (hunk 12 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/authenticator/AuthenticatorCredentials.java]: done

File: src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java (hunk 13 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java]: removed

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java (hunk 14 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: removed

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 16 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:101]: done

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 17 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: removed

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java [hunk 18]
```diff
@@ -19,11 +19,13 @@
 import com.sforce.soap.partner.fault.IApiFault;
 import com.sforce.ws.ConnectionException;
 import com.sforce.ws.ConnectorConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
 import io.cdap.plugin.salesforce.authenticator.Authenticator;
 import io.cdap.plugin.salesforce.authenticator.AuthenticatorCredentials;
 import io.cdap.plugin.salesforce.plugin.OAuthInfo;
 import org.apache.hadoop.conf.Configuration;
 
+
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java]: Remove the extra line

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java [hunk 19]
```diff
@@ -89,12 +94,12 @@ public void prepareRun(BatchSinkContext context) {
     if (inputSchema.getFields() != null && !inputSchema.getFields().isEmpty()) {
       String operationDescription = String.format("Wrote to Salesforce %s", config.getSObject());
       lineageRecorder.recordWrite("Write", operationDescription,
-                                  inputSchema.getFields().stream()
-                                    .map(Schema.Field::getName)
-                                    .collect(Collectors.toList()));
+        inputSchema.getFields().stream()
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: This indentation looks incorrect.

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java [hunk 20]
```diff
@@ -89,12 +94,12 @@ public void prepareRun(BatchSinkContext context) {
...
+          .map(Schema.Field::getName)
+          .collect(Collectors.toList()));
     }
     context.addOutput(Output.of(config.getReferenceNameOrNormalizedFQN(orgId, config.getSObject()),
-                                new SalesforceOutputFormatProvider(config)));
+      new SalesforceOutputFormatProvider(config)));
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: Indentation looks incorrect

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java [hunk 21]
```diff
@@ -120,7 +120,7 @@ protected SalesforceBaseSourceConfig(String referenceName,
                                        @Nullable String operation) {
     super(referenceName);
     this.connection = new SalesforceConnectorConfig(consumerKey, consumerSecret, username, password, loginUrl,
-                                                    securityToken, connectTimeout, oAuthInfo);
+      securityToken, connectTimeout, oAuthInfo);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: incorrect indentation. Have a check for all the changes.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 15 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: done

File: src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java (hunk 18 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceConnectionUtil.java]: done

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java (hunk 19 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: done

File: src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java (hunk 20 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceBatchSink.java]: done

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java (hunk 21 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java]: done. Few more files got added to commit when checking for indentation.

File: src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java [hunk 22]
```diff
@@ -64,7 +64,7 @@ public SObjectDescriptor visitStatement(SOQLParser.StatementContext ctx) {
 
       if (!aliasedFields.isEmpty()) {
         throw new SOQLParsingException("Field aliasing is not allowed except when using a GROUP BY clause. "
-          + "List of fields with alias: " + aliasedFields);
+                                         + "List of fields with alias: " + aliasedFields);
```
Comment (greeshmaswaminathan) [inline src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java:67]: What is the context for the changes here ? Is it only formatting? Skip formatting for the code that is not changed for this PR. Hard to see the exact changes otherwise.

Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryVisitor.java:67]: these are just format issues.

**Comment by vikasrathee-cs** [comment]
Closing this PR due to merge conflicts. Raised the PR for same commit https://github.com/data-integrations/salesforce/pull/166
